src/mon/MonmapMonitor.cc

```diff
@@ -308,8 +308,10 @@ bool MonmapMonitor::preprocess_command(MonOpRequestRef op)
       rdata.append(ds);
       ss << "dumped monmap epoch " << p->get_epoch();
     }
-    if (p != mon->monmap)
+    if (p != mon->monmap) {
        delete p;
+       p = nullptr;
```

Comment:
i think clang analyzer will complain at seeing this: assign to a variable which will never be read.

Reply:
@shinobu-x ping?